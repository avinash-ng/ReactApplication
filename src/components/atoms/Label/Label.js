import React from 'react'
import PropTypes from 'prop-types'
const Label = (props) => {
    return (
        <span className="a__label" >{props.label}</span>
        
    )
}

Label.propTypes = {
    label : PropTypes.string.isRequired
}
export default Label