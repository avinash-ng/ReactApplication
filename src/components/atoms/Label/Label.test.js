import Label from './Label'
import expect from 'expect'
import sinon from 'sinon';
import { shallow, configure } from 'enzyme'
import React from 'react'
import Adapter from 'enzyme-adapter-react-16';
// import renderer from 'react-test-renderer'
import { render } from '@testing-library/react';

configure({ adapter: new Adapter() });
describe('<Label /> ', () => {
    it('renders Label element', () => {
        const RenderComponent = shallow(
            <Label label="email"></Label>
        );
        // console.log(RenderComponent)
        expect(RenderComponent.exists()).toBe(true)
    })

    it('recieved props correctly', () => {
        const label = "First Name"
        
        const RenderComponent = shallow(<Label label={label}></Label>)
        expect(RenderComponent.find('span').text()).toBe(label)

    })


    it('Matches snapshot', () => {
            const label="First Name"
            const RenderComponent = shallow(<Label label={label}></Label>)
            expect(RenderComponent).toMatchSnapshot()
    
    })
})