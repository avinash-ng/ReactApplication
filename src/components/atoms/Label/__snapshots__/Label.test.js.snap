// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Label />  Matches snapshot 1`] = `ShallowWrapper {}`;
